doc/threadsafe_function_ex.md

```diff
@@ -0,0 +1,243 @@
+function. A thread should call this API when it stops making use of this
+thread-safe function. Using any thread-safe APIs after having called this API
+has undefined results in the current thread, as it may have been destroyed.
+
+```cpp
+napi_status Napi::ThreadSafeFunctionEx<ContextType, DataType, Callback>::Release()
+```
+
+Returns one of:
+- `napi_ok`: The thread-safe function has been successfully released.
+- `napi_invalid_arg`: The thread-safe function's thread-count is zero.
+- `napi_generic_failure`: A generic error occurred when attemping to release the
+  thread-safe function.
+
+### Abort
+
+"Abort" the thread-safe function. This will cause all subsequent APIs associated
+with the thread-safe function except `Release()` to return `napi_closing` even
+before its reference count reaches zero. In particular, `BlockingCall` and
+`NonBlockingCall()` will return `napi_closing`, thus informing the threads that
+it is no longer possible to make asynchronous calls to the thread-safe function.
+This can be used as a criterion for terminating the thread. Upon receiving a
+return value of `napi_closing` from a thread-safe function call a thread must
+make no further use of the thread-safe function because it is no longer
+guaranteed to be allocated.
+
+```cpp
+napi_status Napi::ThreadSafeFunctionEx<ContextType, DataType, Callback>::Abort()
+```
+
+Returns one of:
+- `napi_ok`: The thread-safe function has been successfully aborted.
+- `napi_invalid_arg`: The thread-safe function's thread-count is zero.
+- `napi_generic_failure`: A generic error occurred when attemping to abort the
+  thread-safe function.
+
+### BlockingCall / NonBlockingCall
+
+Calls the Javascript function in either a blocking or non-blocking fashion.
+- `BlockingCall()`: the API blocks until space becomes available in the queue.
+  Will never block if the thread-safe function was created with a maximum queue
+  size of `0`.
+- `NonBlockingCall()`: will return `napi_queue_full` if the queue was full,
+  preventing data from being successfully added to the queue.
+
+```cpp
+napi_status Napi::ThreadSafeFunctionEx<ContextType, DataType, Callback>::BlockingCall(DataType* data = nullptr) const
+
+napi_status Napi::ThreadSafeFunctionEx<ContextType, DataType, Callback>::NonBlockingCall(DataType* data = nullptr) const
+```
+
+- `[optional] data`: Data to pass to the callback which was passed to
+  `ThreadSafeFunctionEx::New()`.
+- `[optional] callback`: C++ function that is invoked on the main thread. The
+  callback receives the `ThreadSafeFunction`'s JavaScript callback function to
+  call as an `Napi::Function` in its parameters and the `DataType*` data pointer
+  (if provided). Must implement `void operator()(Napi::Env env, Function
+  jsCallback, DataType* data)`, skipping `data` if not provided. It is not
+  necessary to call into JavaScript via `MakeCallback()` because N-API runs
+  `callback` in a context appropriate for callbacks.
```

Comment:
I think this needs to be removed as it is not a parameter for the Ex version.

Reply:
Absolutely. Was an oversight. This has been addressed [here](https://github.com/nodejs/node-addon-api/pull/742/files#diff-3c83591b72a5c006a3ff969a41d5dc61R172-R173).